hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -376,22 +376,37 @@ void initializeAndPublishProxies() {
             String name = entry.getKey();
```

Comment:
Maybe we can have 1 method for all cases to prevent code duplication, something like this:
```
public void initializeProxies(boolean publishAfterInit) {
        Stream<Map.Entry<String, DistributedObjectFuture>> entryStream = proxies.entrySet().stream()
                .filter(entry -> !entry.getValue().isSetAndInitialized());

        if (publishAfterInit) {
            entryStream.forEach(this::initializeAndPublishProxy);
        } else {
            entryStream.forEach(this::initializeProxy);
        }
    }
```
And we can delete `initializeAndPublishProxies` method.

Reply:
thank you :) last commit addresses it